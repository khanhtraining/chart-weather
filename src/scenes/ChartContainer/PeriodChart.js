import React, { useLayoutEffect, useRef } from 'react'
import { select } from 'd3'
import * as d3 from 'd3'

const PeriodChart = (props) => {
    const ref = useRef()
    const { width, height, data } = props
    const x = d3.scalePoint()
        .domain(((data).map(function (d) { return d.hour })))
    const y = d3.scaleLinear()
        .domain([0, 3])
    x.range([0, width])
    y.range([height, 0])

    useLayoutEffect(() => {
        const g = select(ref.current)
        const drawRectInformationChart = () => {
            g.append('rect')
                .data(([data]))
                .attr("class", 'rect-infor')
                .attr('x', function (d) { return x(d[0]?.hour) })
                .attr("y", function (d) { return y(d[0]?.tide) })

            g.append('rect')
                .data(([data]))
                .attr("class", 'rect-infor')
                .attr('x', function (d) { return x(d[1]?.hour) })
                .attr("y", function (d) { return y(d[1]?.tide) })

            g.append('rect')
                .data(([data]))
                .attr("class", 'rect-infor')
                .attr('x', function (d) { return x(d[2]?.hour) })
                .attr("y", function (d) { return y(d[2]?.tide) })

            g.append('rect')
                .data(([data]))
                .attr("class", 'rect-infor')
                .attr('x', function (d) { return x(d[3]?.hour) })
                .attr("y", function (d) { return y(d[3]?.tide) })

            g.append('rect')
                .data(([data]))
                .attr("class", 'rect-infor')
                .attr('x', function (d) { return x(d[4]?.hour) })
                .attr("y", function (d) { return y(d[4]?.tide) })

            g.append('rect')
                .data(([data]))
                .attr("class", 'rect-infor')
                .attr('x', function (d) { return x(d[5]?.hour) })
                .attr("y", function (d) { return y(d[5]?.tide) })

            g.append('rect')
                .data(([data]))
                .attr("class", 'rect-infor')
                .attr('x', function (d) { return x(d[6]?.hour) })
                .attr("y", function (d) { return y(d[6]?.tide) })

            g.append('rect')
                .data(([data]))
                .attr("class", 'rect-infor')
                .attr('x', function (d) { return x(d[7]?.hour) })
                .attr("y", function (d) { return y(d[7]?.tide) })

            g.append('rect')
                .data(([data]))
                .attr("class", 'rect-infor')
                .attr('x', function (d) { return x(d[8]?.hour) })
                .attr("y", function (d) { return y(d[8]?.tide) })

            g.append('rect')
                .data(([data]))
                .attr("class", 'rect-infor')
                .attr('x', function (d) { return x(d[9]?.hour) })
                .attr("y", function (d) { return y(d[9]?.tide) })

            g.append('rect')
                .data(([data]))
                .attr("class", 'rect-infor')
                .attr('x', function (d) { return x(d[10]?.hour) })
                .attr("y", function (d) { return y(d[10]?.tide) })

            g.append('rect')
                .data(([data]))
                .attr("class", 'rect-infor')
                .attr('x', function (d) { return x(d[11]?.hour) })
                .attr("y", function (d) { return y(d[11]?.tide) })

            g.append('rect')
                .data(([data]))
                .attr("class", 'rect-infor')
                .attr('x', function (d) { return x(d[12]?.hour) })
                .attr("y", function (d) { return y(d[12]?.tide) })

            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_hour')
                .attr('x', function (d) { return x(d[0]?.hour) })
                .attr("y", function (d) { return y(d[0]?.tide) })
                .text(function (d) { return d[0].hour })
            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_tide')
                .attr('x', function (d) { return x(d[0]?.hour) })
                .attr("y", function (d) { return y(d[0]?.tide) })
                .text(function (d) { return d[0].tide })
            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_tide--m')
                .attr('x', function (d) { return x(d[0]?.hour) })
                .attr("y", function (d) { return y(d[0]?.tide) })
                .text(`m`)

            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_hour')
                .attr('x', function (d) { return x(d[1]?.hour) })
                .attr("y", function (d) { return y(d[1]?.tide) })
                .text(function (d) { return d[1].hour })
            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_tide')
                .attr('x', function (d) { return x(d[1]?.hour) })
                .attr("y", function (d) { return y(d[1]?.tide) })
                .text(function (d) { return d[1].tide })
            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_tide--m')
                .attr('x', function (d) { return x(d[1]?.hour) })
                .attr("y", function (d) { return y(d[1]?.tide) })
                .text(`m`)

            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_hour')
                .attr('x', function (d) { return x(d[2]?.hour) })
                .attr("y", function (d) { return y(d[2]?.tide) })
                .text(function (d) { return d[2].hour })
            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_tide')
                .attr('x', function (d) { return x(d[2]?.hour) })
                .attr("y", function (d) { return y(d[2]?.tide) })
                .text(function (d) { return d[2].tide })
            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_tide--m')
                .attr('x', function (d) { return x(d[2]?.hour) })
                .attr("y", function (d) { return y(d[2]?.tide) })
                .text(`m`)

            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_hour')
                .attr('x', function (d) { return x(d[3]?.hour) })
                .attr("y", function (d) { return y(d[3]?.tide) })
                .text(function (d) { return d[3].hour })
            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_tide')
                .attr('x', function (d) { return x(d[3]?.hour) })
                .attr("y", function (d) { return y(d[3]?.tide) })
                .text(function (d) { return d[3].tide })
            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_tide--m')
                .attr('x', function (d) { return x(d[3]?.hour) })
                .attr("y", function (d) { return y(d[3]?.tide) })
                .text(`m`)

            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_hour')
                .attr('x', function (d) { return x(d[4]?.hour) })
                .attr("y", function (d) { return y(d[4]?.tide) })
                .text(function (d) { return d[4].hour })
            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_tide')
                .attr('x', function (d) { return x(d[4]?.hour) })
                .attr("y", function (d) { return y(d[4]?.tide) })
                .text(function (d) { return d[4].tide })
            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_tide--m')
                .attr('x', function (d) { return x(d[4]?.hour) })
                .attr("y", function (d) { return y(d[4]?.tide) })
                .text(`m`)

            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_hour')
                .attr('x', function (d) { return x(d[5]?.hour) })
                .attr("y", function (d) { return y(d[5]?.tide) })
                .text(function (d) { return d[5].hour })
            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_tide')
                .attr('x', function (d) { return x(d[5]?.hour) })
                .attr("y", function (d) { return y(d[5]?.tide) })
                .text(function (d) { return d[5].tide })
            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_tide--m')
                .attr('x', function (d) { return x(d[5]?.hour) })
                .attr("y", function (d) { return y(d[5]?.tide) })
                .text(`m`)

            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_hour')
                .attr('x', function (d) { return x(d[6]?.hour) })
                .attr("y", function (d) { return y(d[6]?.tide) })
                .text(function (d) { return d[6].hour })
            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_tide')
                .attr('x', function (d) { return x(d[6]?.hour) })
                .attr("y", function (d) { return y(d[6]?.tide) })
                .text(function (d) { return d[6].tide })
            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_tide--m')
                .attr('x', function (d) { return x(d[6]?.hour) })
                .attr("y", function (d) { return y(d[6]?.tide) })
                .text(`m`)

            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_hour')
                .attr('x', function (d) { return x(d[7]?.hour) })
                .attr("y", function (d) { return y(d[7]?.tide) })
                .text(function (d) { return d[7].hour })
            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_tide')
                .attr('x', function (d) { return x(d[7]?.hour) })
                .attr("y", function (d) { return y(d[7]?.tide) })
                .text(function (d) { return d[7].tide })
            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_tide--m')
                .attr('x', function (d) { return x(d[7]?.hour) })
                .attr("y", function (d) { return y(d[7]?.tide) })
                .text(`m`)

            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_hour')
                .attr('x', function (d) { return x(d[8]?.hour) })
                .attr("y", function (d) { return y(d[8]?.tide) })
                .text(function (d) { return d[8].hour })
            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_tide')
                .attr('x', function (d) { return x(d[8]?.hour) })
                .attr("y", function (d) { return y(d[8]?.tide) })
                .text(function (d) { return d[8].tide })
            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_tide--m')
                .attr('x', function (d) { return x(d[8]?.hour) })
                .attr("y", function (d) { return y(d[8]?.tide) })
                .text(`m`)

            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_hour')
                .attr('x', function (d) { return x(d[9]?.hour) })
                .attr("y", function (d) { return y(d[9]?.tide) })
                .text(function (d) { return d[9].hour })
            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_tide')
                .attr('x', function (d) { return x(d[9]?.hour) })
                .attr("y", function (d) { return y(d[9]?.tide) })
                .text(function (d) { return d[9].tide })
            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_tide--m')
                .attr('x', function (d) { return x(d[9]?.hour) })
                .attr("y", function (d) { return y(d[9]?.tide) })
                .text(`m`)

            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_hour')
                .attr('x', function (d) { return x(d[10]?.hour) })
                .attr("y", function (d) { return y(d[10]?.tide) })
                .text(function (d) { return d[10].hour })
            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_tide')
                .attr('x', function (d) { return x(d[10]?.hour) })
                .attr("y", function (d) { return y(d[10]?.tide) })
                .text(function (d) { return d[10].tide })
            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_tide--m')
                .attr('x', function (d) { return x(d[10]?.hour) })
                .attr("y", function (d) { return y(d[10]?.tide) })
                .text(`m`)

            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_hour')
                .attr('x', function (d) { return x(d[11]?.hour) })
                .attr("y", function (d) { return y(d[11]?.tide) })
                .text(function (d) { return d[11].hour })
            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_tide')
                .attr('x', function (d) { return x(d[11]?.hour) })
                .attr("y", function (d) { return y(d[11]?.tide) })
                .text(function (d) { return d[11].tide })
            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_tide--m')
                .attr('x', function (d) { return x(d[11]?.hour) })
                .attr("y", function (d) { return y(d[11]?.tide) })
                .text(`m`)

            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_hour')
                .attr('x', function (d) { return x(d[12]?.hour) })
                .attr("y", function (d) { return y(d[12]?.tide) })
                .text(function (d) { return d[12].hour })
            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_tide')
                .attr('x', function (d) { return x(d[12]?.hour) })
                .attr("y", function (d) { return y(d[12]?.tide) })
                .text(function (d) { return d[12].tide })
            g.append('text')
                .data(([data]))
                .attr("class", 'text-infor_tide--m')
                .attr('x', function (d) { return x(d[12]?.hour) })
                .attr("y", function (d) { return y(d[12]?.tide) })
                .text(`m`)
        }
        const rectRemove = () => {
            g.selectAll('rect')
                .data([(data)])
                .exit()
                .remove('rect')
        }
        const textRemove = () => {
            g.selectAll('text')
                .data([(data)])
                .exit()
                .remove('text')
        }
        const remove = () => {
            rectRemove()
            textRemove()
        }
        drawRectInformationChart()

        window.addEventListener('resize', remove)
        window.addEventListener('resize', drawRectInformationChart)
        drawRectInformationChart()
    }, [width, height, data])

    return (
        <g className='weather-container' ref={ref}></g>
    )
}

export default PeriodChart
