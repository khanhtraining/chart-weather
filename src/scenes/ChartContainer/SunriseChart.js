import React, { useLayoutEffect, useRef } from 'react'
import { select } from 'd3'
import * as d3 from 'd3'

const SunriseChart = (props) => {
    const ref = useRef()
    const { width, height, data } = props
    const x = d3.scalePoint()
        .domain(((data).map(function (d) { return d.hour })))
    const y = d3.scaleLinear()
        .domain([0, 3])
    x.range([0, width])
    y.range([height, 0])

    useLayoutEffect(() => {
        const g = select(ref.current)
        const drawRectSunRise = () => {
            g.append('rect')
                .data(([data]))
                .attr("class", 'rect-line')
                .attr('x', function (d) { return x(d[0]?.hour) })
                .attr("y", function (d) { return y(d[0]?.tide) })
                .attr("width", width / 12)
                .attr("height", height * 4)
                .attr("transform", `translate(0,${height - height * 3})`)
            g.append('rect')
                .data(([data]))
                .attr("class", 'rect-line')
                .attr('x', function (d) { return x(d[3]?.hour) })
                .attr("y", function (d) { return y(d[3]?.tide) })
                .attr("width", width / 12)
                .attr("height", height * 4)
                .attr("transform", `translate(0,${height - height * 3})`)
            g.append('rect')
                .data(([data]))
                .attr("class", 'rect-line')
                .attr('x', function (d) { return x(d[4]?.hour) })
                .attr("y", function (d) { return y(d[4]?.tide) })
                .attr("width", width / 12)
                .attr("height", height * 4)
                .attr("transform", `translate(0,${height - height * 3})`)
            g.append('rect')
                .data(([data]))
                .attr("class", 'rect-line')
                .attr('x', function (d) { return x(d[7]?.hour) })
                .attr("y", function (d) { return y(d[7]?.tide) })
                .attr("width", width / 12)
                .attr("height", height * 4)
                .attr("transform", `translate(0,${height - height * 3})`)
            g.append('rect')
                .data(([data]))
                .attr("class", 'rect-line')
                .attr('x', function (d) { return x(d[8]?.hour) })
                .attr("y", function (d) { return y(d[8]?.tide) })
                .attr("width", width / 12)
                .attr("height", height * 4)
                .attr("transform", `translate(0,${height - height * 3})`)
            g.append('rect')
                .data(([data]))
                .attr("class", 'rect-line')
                .attr('x', function (d) { return x(d[11]?.hour) })
                .attr("y", function (d) { return y(d[11]?.tide) })
                .attr("width", width / 12)
                .attr("height", height * 4)
                .attr("transform", `translate(0,${height - height * 3})`)
            g.append('rect')
                .data(([data]))
                .attr("class", 'rect-line')
                .attr('x', function (d) { return x(d[12]?.hour) })
                .attr("y", function (d) { return y(d[12]?.tide) })
                .attr("width", width / 12)
                .attr("height", height * 4)
                .attr("transform", `translate(0,${height - height * 3})`)
        }
        const lineRemove = () => {
            g.selectAll('rect')
                .data([(data)])
                .exit()
                .remove('rect')
        }
        drawRectSunRise()
        window.addEventListener('resize', lineRemove)
        window.addEventListener('resize', drawRectSunRise)
        drawRectSunRise()
    }, [width, height, data])

    return (
        <g className='weather-container' ref={ref}></g>
    )
}

export default SunriseChart
