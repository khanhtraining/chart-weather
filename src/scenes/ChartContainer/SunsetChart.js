import React, { useEffect, useRef } from 'react'
import { select } from 'd3'
import * as d3 from 'd3'

const SunsetChart = (props) => {
    const ref = useRef()
    const { width, height, data } = props
    const x = d3.scalePoint()
        .domain(((data).map(function (d) { return d.hour })))
        .range([0, width])
    const y = d3.scaleLinear()
        .domain([0, 3])
        .range([height, 0])

    useEffect(() => {
        const g = select(ref.current)
        const lineSunPath = d3.area()
            .x(function (d) { return x(d.hour) })
            .y(function (d) { return y(d.sun) })
            .curve(d3.curveMonotoneX)
        g.append('path')
            .data([(data)])
            .attr('class', 'line-sun')
            .style("stroke", 'orange')
            .style("stroke-width", '2px')
            .style("fill", 'none')
            .attr('d', lineSunPath)
    }, [width, height, data, x, y])

    return (
        <g className='sunset-chart' ref={ref} />
    )
}

export default SunsetChart
