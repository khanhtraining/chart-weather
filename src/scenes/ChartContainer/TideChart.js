import React, { useLayoutEffect, useRef } from 'react'
import { select } from 'd3'
import * as d3 from 'd3'

const TideChart = (props) => {
    
    const ref = useRef()
    const { width, height, data } = props
    const x = d3.scalePoint()
        .domain(((data).map(function (d) { return d.hour })))
    const y = d3.scaleLinear()
        .domain([0, 3])
    x.range([0, width])
    y.range([height, 0])

    useLayoutEffect(() => {
        const g = select(ref.current)
        const drawTideChart = () => {
            const area = d3.area()
                .x(function (d) { return x(d.hour) })
                .y0(height)
                .y1((function (d) { return y(d.tide) }))
                .curve(d3.curveBumpX)

            g.append('path')
                .datum((data))
                .attr('class', 'areaTide')
                .attr('d', area((data)))
                .attr('fill', '#0affe840')
        }

        const lineRemove = () => {
            g.selectAll('path')
                .data([(data)])
                .exit()
                .remove('path')
        }

        drawTideChart()
        window.addEventListener('resize', lineRemove)
        window.addEventListener('resize', drawTideChart)
        drawTideChart()
    }, [width, height, data])

    return (
        <g className='weather-container' ref={ref}></g>
    )
}

export default TideChart
