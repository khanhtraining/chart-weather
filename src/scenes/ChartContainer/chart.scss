.svgchart {
    background-color: #ffffff;
    width: 100%;
    height: 100%;
}
.rect-infor {
    transform: translate(-40px, -50px);
    fill: rgb(178, 179, 177);
    width: 78px;
    height: 50px;
}
.rect-line {
    fill: gray;
    opacity: 0.1;
}
.text-infor_hour {
    fill: black;
    font-size: 11px;
    transform: translate(-33px, -30px);
}
.text-infor_tide {
    fill: black;
    font-size: 14px;
    transform: translate(-15px, -13px);
    &--m {
        transform: translate(5px, -13px);
    }
}
.text-infor_replace {
    transform: translate(-1000px, -1000px);
}
