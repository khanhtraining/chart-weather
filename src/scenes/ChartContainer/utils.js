import * as d3 from 'd3'

const getXYAxis = (data, width, height) => {
    const x = d3.scalePoint()
        .domain(data.map(d => d.hour))
    const y = d3.scaleLinear()
        .domain([0, 3])
    x.range([0, width])
    y.range([height, 0])
    return { x, y }
}

export {
    getXYAxis
}
