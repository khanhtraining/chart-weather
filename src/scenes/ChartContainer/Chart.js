import React, { useEffect } from 'react'
import './chart.scss'
import * as d3 from 'd3'
<script src="https://d3js.org/d3.v4.min.js" charset="utf-8"></script>

export const Chart = (props) => {
    useEffect(() => {
        var Svg = d3.select(".svgchart")
            .append("svg")
        var g = Svg.append("g")
            .attr('class', 'weather-container')
        var x = d3.scalePoint()
            .domain(((props.chartData).map(function (d) { return d.hour; })))
        var y = d3.scaleLinear()
            .domain([0, 3])
        var xAxis = g.append("g")
            .style('color', '#000')
        var yAxis = g.append("g")
        var lineSunColor = 'orange'

        var lineTidePath = d3.area()
            .x(function (d) { return x(d.hour); })
            .y(function (d) { return y(d.tide); })
            .curve(d3.curveBumpX),
            lineSunPath = d3.area()
                .x(function (d) { return x(d.hour); })
                .y(function (d) { return y(d.sun); })
                .curve(d3.curveMonotoneX);

        const rectangle = () => {
            g.append('path')
                .data((props.chartData))
                .attr("class", 'rect-line')
                .attr("x", function (d) { return x(d.hour); })
                .attr("y", function (d) { return y(d.tide); })
                .attr("width", 50)
                .attr("height", 30)
                .style("stroke", 'pink')
                .style("fill", 'pink')
                .style("stroke-width", '1px')
        }

        const lineTideMain = () => {
            g.append('path')
                .data([(props.chartData)])
                .attr('class', 'line-tide')
                .style("stroke", '#0affe840')
                .attr('d', lineTidePath)
        }
        const lineSunMain = () => {
            g.append('path')
                .data([(props.chartData)])
                .attr('class', 'line-sun')
                .style("stroke", lineSunColor)
                .style("stroke-width", '2px')
                .style("fill", 'none')
                .attr('d', lineSunPath)
        }
        const lineRemove = () => {
            g.selectAll('path')
                .data([(props.chartData)])
                .exit()
                .remove('path')
        }
        const drawChart = () => {
            const currentWidth = parseInt(d3.select('.svgchart').style('width'), "100%")
            const currentHeight = parseInt(d3.select('.svgchart').style('height'), "100%")
            Svg.attr("width", currentWidth * 6)
                .attr('height', currentHeight)
            x.range([0, currentWidth * 6])
            xAxis.call(d3.axisBottom(x).ticks(2)).attr("transform", `translate(0,${currentHeight - 20})`)
            y.range([currentHeight - 20, 0]);
            yAxis.call(d3.axisLeft(y).ticks(10))
            var area = d3.area()
                .x(function (d) { return x(d.hour); })
                .y0(currentHeight - 20)
                .y1((function (d) { return y(d.tide); }))
                .curve(d3.curveBumpX)

            g.append('path')
                .data((props.chartData))
                .attr('class', 'area')
                .attr('d', area((props.chartData)))
                .attr('fill', '#0affe840')
            lineTideMain()
            lineSunMain()
            rectangle()
        }
        const remove = () => {
            lineRemove()
        }
        drawChart()

        window.addEventListener('resize', remove)
        window.addEventListener('resize', drawChart)
    }, [])

    return (
        <div className='layout__container-chart'>
            <div className='layout__container-chart--title'>
                <span className='blue-title'>Tide</span>
                <span className='orange-title'>Sunrise & Sunset</span>
            </div>
            <div className='svgchart'>
            </div>
        </div>
    )
}

export default Chart
