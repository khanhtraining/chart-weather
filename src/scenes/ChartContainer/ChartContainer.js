import React, { useEffect, useState } from 'react'
import * as d3 from 'd3'
import TideChart from './TideChart'
import SunsetChart from './SunsetChart'
import SunriseChart from './SunriseChart'
import PeriodChart from './PeriodChart'
import './chart.scss'

export const ChartContainer = (props) => {
    const [width, setWidth] = useState(0)
    const [height, setHeight] = useState(0)

    useEffect(() => {
        const currentWidth = parseInt(d3.select('.svgchart').style('width'), "100%")
        const currentHeight = parseInt(d3.select('.svgchart').style('height'), "100%")
        setWidth(currentWidth * 6)
        setHeight(currentHeight)
    }, [width, height, props.chartData])

    return (
        <div className='layout__container-chart'>
            <div className='svgchart'>
                <svg width={width} height={height}>
                    <TideChart width={width} height={height} data={props.chartData} />
                    <SunsetChart width={width} height={height} data={props.chartData} />
                    <PeriodChart width={width} height={height} data={props.chartData} />
                    <SunriseChart width={width} height={height} data={props.chartData} />
                </svg>
            </div>
            <div className='layout__container-chart--title'>
                <span className='blue-title'>Tide â€¢</span>
                <span className='orange-title'>Sunrise &amp; Sunset</span>
            </div>
        </div>
    )
}

export default ChartContainer
