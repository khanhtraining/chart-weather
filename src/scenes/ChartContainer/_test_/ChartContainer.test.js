import React from 'react'
import renderer from 'react-test-renderer'
import ChartContainer from '../ChartContainer'

describe('snapshot testing', () => {
    test('snapshot for chart container component', () => {
        const renderedComponent = renderer.create(<ChartContainer/>).toJSON()
        expect(renderedComponent).toMatchSnapshot()
    })
})
