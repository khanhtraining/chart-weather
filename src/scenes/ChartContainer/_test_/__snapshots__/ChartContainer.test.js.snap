// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot for Today Detail component 1`] = `
<div
  className="layout__container-chart"
>
  <div
    className="svgchart"
  >
    <svg
      height={0}
      width={0}
    >
      <g
        className="tide-chart"
      />
      <g
        className="sunset-chart"
      />
      <g
        className="period-chart"
      />
      <g
        className="sunrise-chart"
      />
    </svg>
  </div>
  <div
    className="layout__container-chart--title"
  >
    <span
      className="blue-title"
    >
      Tide •
    </span>
    <span
      className="orange-title"
    >
      Sunrise & Sunset
    </span>
  </div>
</div>
`;

exports[`snapshot testing snapshot for chart container component 1`] = `
<div
  className="layout__container-chart"
>
  <div
    className="svgchart"
  >
    <svg
      height={0}
      width={0}
    >
      <g
        className="tide-chart"
      />
      <g
        className="sunset-chart"
      />
      <g
        className="period-chart"
      />
      <g
        className="sunrise-chart"
      />
    </svg>
  </div>
  <div
    className="layout__container-chart--title"
  >
    <span
      className="blue-title"
    >
      Tide •
    </span>
    <span
      className="orange-title"
    >
      Sunrise & Sunset
    </span>
  </div>
</div>
`;
